src/Api/Order.php

```diff
@@ -107,13 +113,95 @@ public static function place_order( \WP_REST_Request $request ) {
 		$order_id = $checkout->create_order();
 		$order = new \WC_Order( $order_id );
 
+		do_action( Filters::AFTER_ORDER, $request, $cart );
+
+		// If the user is not logged in, we need to pass the billing and shipping address to the order.
+		if ( ! is_user_logged_in() ) {
+			$order = self::update_guest_order( $request, $order );
+		}
+
 		// Empty the current cart.
 		$cart->empty_cart();
 
 		return $order;
 	}
 
 	/**
+	 * If the user is not logged in, we need the shipping and billing info in order to create the shop order.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @param \WC_Order 	   $order Order.
+	 * @return \WP_Error || \WC_Order
+	 */
+	public static function update_guest_order( \WP_REST_Request $request, $order ) {
+		do_action( Filters::GUEST_PRE_UPDATE_ORDER, $request, $order );
+
+		$params = $request->get_body_params();
+
+		if ( ! self::check_post_address( $params ) ) {
+			return new \WP_Error(
+				ErrorCodes::BAD_REQUEST,
+				'Guest order needs billing and shipping information in POST body.',
+				[ 'status' => HttpCodes::HTTP_BAD_REQUEST ]
+			);
+		}
+
+		if ( self::fields_have_errors( $params ) ) {
+			return new \WP_Error(
+				ErrorCodes::BAD_REQUEST,
+				'Shipping and/or billing address are not completed. Please provide the minimum fields in POST body.',
+				[ 'status' => HttpCodes::HTTP_BAD_REQUEST ]
+			);
+		}
+
+		// Update order with shipping and billing information for the Guest.
+		$order->set_address( $params['shipping'], self::SHIPPING_KEY );
+		$order->set_address( $params['billing'], self::BILLING_KEY );
+
+		do_action( Filters::GUEST_AFTER_UPDATE_ORDER, $request, $order );
+
+		return $order;
+	}
+
+	/**
+	 * Check if the required parameters are on the POST body.
+	 *
+	 * @param array	$params  Post body parameters.
+	 * @return bool
+	 */
+	public static function check_post_address( $params ) {
+		if ( ! array_key_exists( self::SHIPPING_KEY, $params ) || ! array_key_exists( self::BILLING_KEY, $params ) ) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Check if we are receiving the minimum required fields.
+	 *
+	 * @param array	$params  Post body parameters.
+	 * @return int Number of errors.
+	 */
+	public static function fields_have_errors( $params ) {
+		$errors = 0;
+		$required_billing = count( self::BILLING_REQUIRED_FIELDS );
+		$required_shipping = count( self::SHIPPING_REQUIRED_FIELDS );
+
+		$array_billing = array_flip( self::BILLING_REQUIRED_FIELDS );
+		$array_shipping = array_flip( self::SHIPPING_REQUIRED_FIELDS );
```

Comment:
Yes sorry:

```php
// Simplified arrays, is this structure but with more fields of course.
$a = ['billing_address_1', 'billing_email', 'billing_postcode']
$b = array(
    'billing' => array(
        'billing_address'   => '24 Broadway',
        'billing_email'       => 'me@gmail.com',
        'billing_postcode' => '11003',
        'optional_thing'    => 'Some stuff',
     )
)

// 
```
I want to know if all the keys in `$b['billing']` are present in `$a` (we may have more, but we need minimum these ones).

So, in this case, it will be true, becase we have in $b['billing'] all the keys in $a, and 1 more, which is OK.